libs/ml/tests/ml/ops/add.cpp

```diff
@@ -98,4 +187,5 @@ TYPED_TEST(AddTest, backward_test)
                                      fetch::math::function_tolerance<DataType>()));
   ASSERT_TRUE(prediction[1].AllClose(gt_2, fetch::math::function_tolerance<DataType>(),
                                      fetch::math::function_tolerance<DataType>()));
+  ASSERT_TRUE(prediction[1].shape() == data_2.shape());
```

Comment:
This should be tested before AllClose, because if shape is different AllClose would fail.